### ISSUE TITLE
Globale Einstellungen werden nicht übernommen.

### ISSUE THREAD

**Issue by kroerig** [SVWS-NRW/Schild3-BetaTest#851]
Flollowup zu https://github.com/SVWS-NRW/Schild3-BetaTest/issues/799, da nur Colaboratoren Issues wiedereröffnen können.

Die globalen Maileinstellungen werden auch in Version 3.0.78.14 nicht übernommen. In SchILD2 stehen sie auf "Direktes SMTP", nach der Migration stehen sie in SchILD3 auf "Mailprogramm" und die SMTP-Felder sind leer.

In der Datenbank steht:

```
SMTPMail=1
SMTPServer="mailserver"
SMTPPort=25
SMTPSSL=0
SMTPStartTLS=1
```

Aber SchILD3 weiß nichts davon.

**Comment by kroerig** [comment]
Auch `ErzieherBeiVolljaehrigkeitDeaktivieren=1` wird nicht übernommen.

**Comment by kroerig** [comment]
![grafik](https://github.com/SVWS-NRW/Schild3-BetaTest/assets/5148795/14e8ae80-eabf-4ed3-9f3d-97ce0cb26994)
![grafik](https://github.com/SVWS-NRW/Schild3-BetaTest/assets/5148795/ca9cecc0-1a61-48c8-b385-a392c12a64ca)

![grafik](https://github.com/SVWS-NRW/Schild3-BetaTest/assets/5148795/fcd56979-addf-49f4-a2f2-c96e8a9053e4)
![grafik](https://github.com/SVWS-NRW/Schild3-BetaTest/assets/5148795/6c5328a3-6508-4f60-bf4e-6619914184bc)

![grafik](https://github.com/SVWS-NRW/Schild3-BetaTest/assets/5148795/7fe34908-b579-4f4c-8d8c-c44e8fcbadb5)
![grafik](https://github.com/SVWS-NRW/Schild3-BetaTest/assets/5148795/dc507c7e-4b36-4ef4-b1c4-faf9b487e53c)

**Comment by JuergenRichter** [comment]
Da hat sich möglicherweise im VWS-Server was geändert. Wenn ich das im Debugger teste, bekomme ich die Fehlermeldung 400 (bad request), das hat früher funktioniert. Werde ich prüfen.

**Comment by kroerig** [comment]
Ja, nach der Migration scheint SchILD3 nur Defaultwerte anzuzeigen, obwohl die richtigen Einstellungen in der Datenbank stehen.
Testweise habe ich mal in SchILD3 einen Wert geändert: "0" Bei Telefonvorwahl für TK-Anlage eingetragen.
Der Wert kommt in `EigeneSchule.Einstellungen` nicht an, oder ich schaue an der falschen Stelle.

Aber nach der Änderungen konnte ich SchiLD nicht mehr beenden. Und zwar durch diesen Fehler: https://github.com/SVWS-NRW/Schild3-BetaTest/issues/852

**Comment by AnneSchueller** [comment]
In der Tabelle client_configuration_global stehen bei mir trotzdem Werte drin. Allerdings nicht die Werte, die in der Schild2 DB unter eigeneSchule vorliegen.
In der Tabelle eigeneSchule der migrierten MariaDB  stehen die migrierten Werte richtig.

**Comment by JuergenRichter** [comment]
Sobald die Datenbank einmal mit SchILD3 geöffnet wurde, stehen da auch Werte drin, das sind dann deie Defaults. Bitte mal frisch migrieren, danach ist vermutlich die Tabelle client_configuration_global leer.

**Comment by AnneSchueller** [comment]
Stimmt, nach der Migration ist die Tabelle leer.

**Comment by kroerig** [comment]
Aber eigentlich wäre das ja dann Aufgabe des SVWS Servers, die Werte bei der Migration an die richtige Stelle zu übertragen.

Klar könnte Schild3 beim Start schauen, die Tabelle leer ist und, sofern vorhanden, die INI Datei von der alten Stelle übertragen. 

Das muss ja auch dann noch funktionieren, wenn es den Desktopclient nicht mehr braucht.

**Comment by JuergenRichter** [comment]
Ich habe es jetzt erst mal so wie beschreiben umgesetzt

**Comment by kroerig** [comment]
Ich bitte das Issue wieder zu öffnen.

Ausgelöst durch https://github.com/SVWS-NRW/SVWS-Server/issues/250#issuecomment-2378817205 da ich die Einstellungen nachgeschaut hatte, ist mir aufgefallen, dass in der aktuellen Version 3.0.91.15 die Maileinstellungen nicht übernommen werden.

Im SVWS Client wird der SMTP Server richtig angezeigt, die Einstellungen kommen bei der Migration also mit. Stehen auch in `client_configuration_global.Einstellungen` unter AppName=Schild2, Schlüssel=Einstellungen.

Auch die Benutzereinstellungen stehen in der Tabelle `BenutzerEmail`, wenn auch teilweise mit leeren Zeilen.

**Comment by kroerig** [comment]
Auch die anderen Einstellungen stimmen nicht (mehr).

SchILD2:
![grafik](https://github.com/user-attachments/assets/2e2bf22c-7b5c-490b-a196-d4c901bbe10f)

SchILD3:
![grafik](https://github.com/user-attachments/assets/706f8b92-ff13-41a7-8880-6aa5c0c97dd0)

**Comment by JuergenRichter** [comment]
Im nächsten Update

**Comment by AnneSchueller** [comment]
Nach meinen Tests werden die oben genannten Einstellungen nun für Schild3 korrekt übernommen.

Nur das Feld für die Mindestnote beim Holen scheint noch nicht korrigiert zu sein:
![image](https://github.com/user-attachments/assets/339bdb66-1441-4a7d-9e1e-485b138dbe98)

**Comment by kroerig** [comment]
Dann werde ich die Tage nochmal eine Migration machen und schauen.
